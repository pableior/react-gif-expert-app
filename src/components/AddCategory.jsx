
import {useState} from 'react';
import PropTypes from 'prop-types';
export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setinputValue] = useState('');

    const onInputChanged = ({target}) => {
        setinputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length < 1) return;

        //setCategories(categories => [inputValue, ...categories]);
        setinputValue('');
        
        
        onNewCategory(inputValue.trim());


    }

    return (
        <form onSubmit={ (event) => onSubmit(event)} aria-label="form">
            <input
                type="text"
                placeholder="Buscar Gifts"
                value= {inputValue}
                onChange={(event) => onInputChanged(event)}
            />
        </form>
    )
    
}


AddCategory.propTypes  = {
    onNewCategory: PropTypes.func.isRequired
}