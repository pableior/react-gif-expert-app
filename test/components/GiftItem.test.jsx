import { findByAltText, render, screen } from '@testing-library/react';
import { GifItem } from "../../src/components/GifItem";

describe('Tests on <GifItem /> component', () => {


    const title = 'Imagen';
    const url = 'https://domain/path'
    
    test('must match to snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>);



        expect(container).toMatchSnapshot();
    });


    test(('must show  image with url and alternative value indicated'), () => {
        render(<GifItem title={ title } url = { url} />);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });


    test('mmust show title in component', () => { 
        render(<GifItem title={ title } url = { url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })



});