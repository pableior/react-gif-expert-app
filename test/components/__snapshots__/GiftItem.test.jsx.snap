// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem /> component must match to snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Imagen"
      src="https://domain/path"
    />
    <p>
      Imagen
    </p>
  </div>
</div>
`;
